Test/Unit/Helper/ConfigTest.php

```diff
@@ -97,545 +154,638 @@ public function setUp()
                     $this->request
                 ]
             )
-            ->getMock();
+            ->setMethods($methods);
 
-        $this->currentMock->method('getScopeConfig')
-            ->will($this->returnValue($this->scopeConfig));
+        if($enableOriginalConstructor) {
+            $builder->enableOriginalConstructor();
+        } else {
+            $builder->disableOriginalConstructor();
+        }
+
+        if($enableProxyingToOriginalMethods) {
+            $builder->enableProxyingToOriginalMethods();
+        } else {
+            $builder->disableProxyingToOriginalMethods();
+        }
+
+        $this->currentMock = $builder->getMock();
     }
 
     /**
      * @test
+     * @dataProvider getMerchantDashboardUrlProvider
      */
-    public function getStoreVersion()
+    public function getMerchantDashboardUrl($sandboxFlag, $expected)
     {
-        $magentoVersion = '2.2.3';
-
-        $this->productMetadata->expects($this->once())
-            ->method('getVersion')
-            ->will($this->returnValue($magentoVersion));
-
-        $result = $this->productMetadata->getVersion();
+        $this->initCurrentMock(['isSandboxModeSet']);
+        $this->currentMock->expects(self::once())->method('isSandboxModeSet')->willReturn($sandboxFlag);
+        $this->assertEquals($expected, $this->currentMock->getMerchantDashboardUrl());
+    }
 
-        $this->assertEquals($magentoVersion, $result, 'Cannot determine magento version');
+    public function getMerchantDashboardUrlProvider ()
+    {
+        return [
+            [true, BoltConfig::MERCHANT_DASH_SANDBOX],
```

Comment:
Same as the OrderHelper unit tests, maybe this is a non-issue but just commenting on the use of constants declared in the system under test to validate test results.